reporters/GlobalSummaryReporter.js

```diff
@@ -153,23 +160,381 @@ class GlobalSummaryReporter {
     console.log(italic("- Unique liquidations count # of unique sponsors that have been liquidated"));
     console.log(italic("- Collateral & tokens liquidated counts aggregate amounts from all partial liquidations"));
     console.log(italic("- Current collateral liquidated includes any collateral locked in pending liquidations"));
-    await this._generateLiquidationStats();
+    await this._generateLiquidationStats(periods);
     console.groupEnd();
 
     // 4. Dispute stats table
     console.group();
     console.log(bold("Dispute summary stats"));
-    await this._generateDisputeStats();
+    await this._generateDisputeStats(periods);
     console.groupEnd();
 
     // 5. DVM stats table
     console.group();
     console.log(bold("DVM summary stats"));
-    await this._generateDvmStats();
+    await this._generateDvmStats(periods);
     console.groupEnd();
   }
 
-  async _generateSponsorStats() {
+  /** *******************************************************
+   *
+   * Helper methods to sort event data by block timestamp
+   *
+   * *******************************************************/
+  async _filterNewSponsorData(periods, newSponsorEvents) {
```

Comment:
These are the refactored event querying methods, they all look the same. They will check event data against all of the input periods.